components/calendar/src/iso.rs

```diff
@@ -183,8 +183,8 @@ impl Calendar for Iso {
             }
         };
 
-        let january_1_2000 = 5; // Saturday
-        let day_offset = (january_1_2000 + year_offset + month_offset + date.day.0 as i32) % 7;
+        let march_1_200 = 5; // Saturday
```

Comment:
suggestion: I don't think `day_of_week` needs to be changed: For other calendars we should just use a default implementation of converting to Iso and calling `day_of_week` on it. For Julian I think the following things need to be different from gregorian

 - code for conversion to/from iso
 - code for handling offsets and subtraction. This should be shared between calendars as already mentioned

Reply:
This doesn't really change the implementation right?! It just changes the date used to measure, how will that effect anything elsewhere? 🤔 

Julian and Georgian are on the same day this day, so it kinda makes things convenient

Reply:
Well you have to introduce the whole slack-calculation step as well.

It's far more straightfoward to keep this impl as is and have Julian (and all the other calendars) just convert to ISO before calling this